### PR TITLE
OHM-829 Add back URL rewriting functionality

### PR DESCRIPTION
**Pull request by Zooloo2014**
URL rewriting permits the HIM to rewrite URLs that occur in the response body before passing the body back up to the invoking client.

### CODE REVIEW

File: src/middleware/router.js [hunk 1]
```diff
@@ -501,9 +497,19 @@ async function sendHttpRequest (ctx, route, options) {
     responseProgress: function (chunk, counter, size) {
       logger.info(`Write response CHUNK # ${counter} [ Total size ${size}]`)
     },
-    finishResponse: function () {
+    finishResponse: function (response, size) {
```
Comment (BMartinos) [inline src/middleware/router.js]: It doesnt seem like we make use of these parameters. Are they needed?

File: src/middleware/router.js [hunk 2]
```diff
@@ -501,9 +497,19 @@ async function sendHttpRequest (ctx, route, options) {
...
       logger.info(`** END OF OUTPUT STREAM **`)
     },
+    finishResponseAsString: function(body) {
```
Comment (BMartinos) [inline src/middleware/router.js]: very minor: just needs a space between "function" and the bracket

File: test/unit/rewriteUrlsTest.js [hunk 3]
```diff
@@ -8,7 +8,7 @@ import * as rewriteUrls from '../../src/middleware/rewriteUrls'
 import * as utils from '../../src/utils'
 
 // TODO: OHM-699 remove the x prepended to describe when url rewriting is back in support
```
Comment (BMartinos) [inline test/unit/rewriteUrlsTest.js]: Lets remove this comment as well. Now that the logic has been added in, this no longer applies

File: src/middleware/router.js (hunk 1 above)
Comment (Zooloo2014) [reply src/middleware/router.js]: These parameters are passed in the call. They may or may not be used in the event handler. Some of the handlers have no code; the parameters are just included to document the what information is available in the call...

Comment (BMartinos) [reply src/middleware/router.js]: Ok cool :+1

**Pull request merged**